aziotctl/src/internal/check/checks/host_connect_iothub.rs

```diff
@@ -159,3 +166,81 @@ impl HostConnectIotHub {
         Ok(CheckResult::Ok)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::internal::check::{CheckerCache, CheckerCfg, CheckerShared, DaemonConfigs};
+
+    use super::HostConnectIotHub;
+    use aziot_identityd_config::{
+        Endpoints, ManualAuthMethod, Provisioning, ProvisioningType, Settings,
+    };
+    use std::path::PathBuf;
+
+    #[tokio::test]
+    async fn tls_handshake_timesout() {
```

Comment:
Please don't make a test that stalls for 10s. It breaks the dev workflow.

Reply:
I have set the cloud timeout as 1 second for this particular test

Reply:
Compiling aziotctl v1.3.0 (/workspaces/iot-identity-service/aziotctl)
    Finished test [unoptimized + debuginfo] target(s) in 13.28s
     Running unittests (/workspaces/iot-identity-service/target/debug/deps/aziotctl-833ade0ec64079e5)

running 1 test
test internal::check::checks::host_connect_iothub::tests::tls_handshake_timesout ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.00s

Reply:
Ah right. Sorry.

I would also like it to not make a network call and expect it to fail. We do have precedence for it with the mini-sntp tests, but at least that's legitimate traffic to NTP servers, unlike connecting to bing.com on port 8883. In fact, it's lucky that (presumably) Bing happens to black-hole the traffic and thus your tests time out, as opposed to it rejecting the connection which would result in a different error. There's no guarantee it couldn't change in the future.

I suggest spinning up a bespoke TCP server (just a spawned task that does `tokio::TcpListener::bind(("::", 0))` + `listener.accept()` + `tokio::time::sleep(5)`) and connecting to that (`"localhost"` for the hostname and `listener.local_addr().port()` for the port).

Reply:
@arsing  : Thanks for the suggestion!, I ended up using just binding to a Port without having to accept connections to keep things simple. I observed that I still get a timeout without having to do ```listener.accept()```. Let me know what you think?

Reply:
Oh right, that'll work too, and yeah that's much easier than spawning a sleeping task.